src/main/java/com/powsybl/openloadflow/network/action/LfActionUtils.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2025, Coreso SA (https://www.coreso.eu/) and TSCNET Services GmbH (https://www.tscnet.eu/)
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.openloadflow.network.action;
+
+import com.powsybl.action.*;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.openloadflow.graph.GraphConnectivity;
+import com.powsybl.openloadflow.network.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+import static com.powsybl.openloadflow.network.action.AbstractLfBranchAction.updateBusesAndBranchStatus;
+
+/**
+ * @author Bertrand Rix {@literal <bertrand.rix at artelys.com>}
+ * @author Anne Tilloy {@literal <anne.tilloy at rte-france.com>}
+ * @author Jean-Luc Bouchot {@literal <jlbouchot at gmail.com>}
+ */
+public final class LfActionUtils {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LfActionUtils.class);
+
+    private LfActionUtils() {
+    }
+
+    public static LfAction createLfAction(Action action, Network network, boolean breakers, LfNetwork lfNetwork) {
```

Comment:
I added the Breaking change tag, as your refactor changed the API.

Add a note about this change and how to migrate in the PR description

Reply:
Not sure about considering it a breaking change, I think it is adapted when changing a public API or the interpretation of an existing input. 
This PR is only an internal change of implementation within OLF, iidm Action is the public API and it stays the same.